packages/cli/src/modules/provisioning.ee/provisioning.service.ee.ts

```diff
@@ -34,6 +41,43 @@ export class ProvisioningService {
 		return this.provisioningConfig;
 	}
 
+	async provisionInstanceRoleForUser(user: User, role: unknown) {
+		let parsedRole: Role;
+
+		try {
+			parsedRole = roleSchema.parse(role);
+		} catch (error) {
+			this.logger.warn(
+				`Invalid role: ${role} provided, expected oneof ${Object.values(ROLE).join(', ')}, skipping instance role provisioning`,
+				{ userId: user.id, role, error },
+			);
+			return;
+		}
+
+		/*
+		 * If the user is changing from an owner to a non-owner role,
+		 * we need to check if they are the last owner to avoid an instance losing its only owner
+		 */
+		if (user.role.slug === ROLE.Owner && parsedRole !== ROLE.Owner) {
+			const otherOwners = await this.userRepository.count({
+				where: { role: { slug: ROLE.Owner }, id: Not(user.id) },
+			});
+
+			if (otherOwners === 0) {
+				this.logger.warn(
+					`Cannot remove last owner role: ${ROLE.Owner} from user: ${user.id}, skipping instance role provisioning`,
+					{ userId: user.id, role },
+				);
+				return;
+			}
+		}
+
+		// No need to update record if the role hasn't changed
+		if (user.role.slug !== parsedRole) {
+			await this.userRepository.update(user.id, { role: { slug: parsedRole } });
```

Comment:
Nit code style: Having reached the end of this method, I was ready to see another "early return" for the "role unchanged" condition, and then a final `this.userRepository.update` after all previous early returns did not "trigger".